boards/stm32f746g-disco/include/periph_conf.h

```diff
@@ -430,6 +431,43 @@ static const fmc_bank_conf_t fmc_bank_config[] = {
 #define FMC_BANK_NUMOF  ARRAY_SIZE(fmc_bank_config)
 /** @} */
 
+/**
+ * @name SDIO/SDMMC configuration
+ * @{
+ */
+
+/**
+ * @brief SDIO/SDMMC static configuration struct
+ */
+static const sdmmc_conf_t sdmmc_config[] = {
+    {
+        .dev = SDMMC1,
+        .bus = APB2,
+        .rcc_mask = RCC_APB2ENR_SDMMC1EN,
+        .cd = GPIO_PIN(PORT_C, 13),
+        .cd_active = 0,             /* CD pin is LOW active */
+        .cd_mode = GPIO_IN_PU,      /* Pull-up R12 not soldered by default */
+        .clk = { GPIO_PIN(PORT_C, 12), GPIO_AF12 },
+        .cmd = { GPIO_PIN(PORT_D,  2), GPIO_AF12 },
+        .dat0 = { GPIO_PIN(PORT_C,  8), GPIO_AF12 },
+        .dat1 = { GPIO_PIN(PORT_C,  9), GPIO_AF12 },
+        .dat2 = { GPIO_PIN(PORT_C, 10), GPIO_AF12 },
+        .dat3 = { GPIO_PIN(PORT_C, 11), GPIO_AF12 },
+#if IS_USED(MODULE_PERIPH_DMA)
+        .dma = 5,
+        .dma_chan = 4,
+#endif
+        .irqn = SDMMC1_IRQn
+    },
+};
+
+/**
+ * @brief Number of configured SDIO/SDMMC peripherals
+ */
+#define SDMMC_CONFIG_NUMOF  1
```

Comment:
Why not use
```suggestion
#define SDMMC_CONFIG_NUMOF  ARRAY_SIZE(sdmmc_config)
```

?

Reply:
Because `ARRAY_SIZE` can't be used in preprocessor conditionals: https://github.com/RIOT-OS/RIOT/blob/075fa497a0db46c1cba946f0e41abfca211399de/cpu/stm32/periph/sdmmc.c#L218 It's needed in the definition of the internal device structure array because the number of the entries in this array depends on the number of the configured devices. To avoid misconfigurations, some sanity checks ensure that `SDMMC_CONFIG_NUMOF` corresponds to `ARRAY_SIZE(sdmmc_config)` and is not greater than the number SDMMMC peripherals provided by the MCU.

Reply:
Alternatively the macro `SDMMC2` could be used here, which is set if the MCU supports 2 SDMMC peripherals. Alternatively the macro `SDMMC2` could be used here, which is set if the MCU supports 2 SDMMC peripherals. But this would have 2 drawbacks:
- two internal devices would be defined in RAM even if only one SDMMC peripheral is configured
- the second internal device would be initialized with `sdmmc_config[1]` that does not exist if only one SDMMC peripheral is configured.

Another problem is that `SDMMC_CONFIG_NUMOF` is used to decide whether a second SDMMC device pointer has to be defined in the XFA of SDMMC devices or not: https://github.com/RIOT-OS/RIOT/blob/075fa497a0db46c1cba946f0e41abfca211399de/cpu/stm32/periph/sdmmc.c#L243-L245 Using `SDMMC2` instead could lead to 2 SDMMC devices without having it configured.

Therefore, I decided to use a constant instead of `ARRAY_SIZE`. I don't think that it is too complicated. The `static_assert` make sure that `SDMMC_CONFIG_NUMOF` matches the number of elements in `sdmmc_config`.